typescript/infra/src/utils/gcloud.ts

```diff
@@ -95,6 +101,53 @@ export async function gcpSecretExists(secretName: string) {
   return matches.length > 0;
 }
 
+/**
+ * Uses the SecretManagerServiceClient to check if a secret exists.
+ * @param secretName The name of the secret to check.
+ * @returns A boolean indicating whether the secret exists.
+ */
+export async function gcpSecretExistsUsingClient(
```

Comment:
similar deal here - imo you're good to just replace the old CLI-using  `gcpSecretExists` entirely with this

Reply:
I'll do this as follow up and take the time to test the code paths manually where I can